src/SFML/Graphics/Text.cpp

```diff
@@ -527,24 +516,31 @@ void Text::ensureGeometryUpdate() const
         // Add the glyph to the vertices
         addGlyphQuad(m_vertices, Vector2f(x, y), m_fillColor, glyph, italicShear);
 
-        // Update the current bounds with the non outlined glyph bounds
-        if (m_outlineThickness == 0)
-        {
-            float left   = glyph.bounds.left;
-            float top    = glyph.bounds.top;
-            float right  = glyph.bounds.left + glyph.bounds.width;
-            float bottom = glyph.bounds.top  + glyph.bounds.height;
-
-            minX = std::min(minX, x + left  - italicShear * bottom);
-            maxX = std::max(maxX, x + right - italicShear * top);
-            minY = std::min(minY, y + top);
-            maxY = std::max(maxY, y + bottom);
-        }
+        // Update the current bounds
+        float left   = glyph.bounds.left;
+        float top    = glyph.bounds.top;
+        float right  = glyph.bounds.left + glyph.bounds.width;
+        float bottom = glyph.bounds.top  + glyph.bounds.height;
+
+        minX = std::min(minX, x + left  - italicShear * bottom);
+        maxX = std::max(maxX, x + right - italicShear * top);
+        minY = std::min(minY, y + top);
+        maxY = std::max(maxY, y + bottom);
 
         // Advance to the next character
         x += glyph.advance + letterSpacing;
     }
 
+    // If we're using outline, update the current bounds
+    if ((m_outlineThickness != 0) && (x > 0))
```

Comment:
I don't think the condition `x > 0` is needed.

Reply:
It actually is so we don't apply outline offset for empty strings.

Reply:
The function exists early (line 421) for empty strings.
I understand the condition you want would be to check if the string is made or whitespace characters only. Should outline be ignored in this case?

Reply:
Sorry I mean strings containing only special symbols tab next line etc. I just followed the other 2 conditions below.

Reply:
Correction, x stays 0 only for `\n` and `\r`

Reply:
I tested and noticed that, before your patch, the bounding box grows when there is some visible outline, i.e. not around whitespace characters. It can happen on one side and not the other, for example if the text starts with whitespace or a blank line.
With your current path, the outline impacts the bounding box around any character unless the text ends with a newline character. This special case looks like a logic error to me. Without the condition `(x > 0)`, the outline impacts the bounding box around any character. That is still changing the behavior compared to before your patch but it looks consistent.

Reply:
Ok I know what you mean. In that case shouldn't we check both string ends for whitespace characters and add outline for each side separately?

Reply:
Guessing precisely where outline appears might be more complex than checking both ends. Note that an sf::Text instance can represent several lines of text as the string can include newline characters.
Maybe the most consistent (and also simple to implement) is to consider that whitespace characters have an invisible outline that impacts the bounding box. It is consistent with the fact that whitespace characters already impact the bounding box even though those characters are invisible.

Reply:
To summarize, I think once we remove the condition `(x > 0)`, this PR will be good to merge.